static/app/components/codecov/container/persistence.tsx

```diff
@@ -0,0 +1,83 @@
+import {captureException} from '@sentry/react';
+
+import type {CodecovContextTypes} from 'sentry/components/codecov/container/container';
+
+// Constants
+const CODECOV_PRODUCT = 'codecov';
+
+// Types
+type CodecovObjectInLocalStorage = {
+  repository: string | null;
+};
+
+// Functions
+/**
+ * Creates Codecov product local storage key signature
+ */
+function makeCodecovLocalStorageKey(orgSlug: string) {
+  return `${CODECOV_PRODUCT}-selection:${orgSlug}`;
+}
+
+/**
+ * Retrieves Codecov data from local storage. It:
+ * 1) Gets data from local storage with the Codecov key.
+ * 2) Attempts to parse the data.
+ * 3) Validates and returns the data.
+ */
+export function getCodecovDataFromLocalStorage(
+  orgSlug: string
+): CodecovObjectInLocalStorage | null {
+  // 1)
```

Comment:
Trying this approach to comment functionality, not sure how I feel but wanted opinions